src/jit/importer.cpp

```diff
@@ -6462,6 +6463,11 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
         call->gtCall.callSig  = new (this, CMK_CorSig) CORINFO_SIG_INFO;
         *call->gtCall.callSig = calliSig;
 #endif // DEBUG
+
+        if (IsTargetAbi(CORINFO_CORERT_ABI))
```

Comment:
That's the only one we shouldn't do that is required for correctness.

There are some other cases of managed `calli` where (depending on the method that has the `calli`) the EE side could say that the expansion is not needed. But it's a perf optimization that we can postpone. On Project N (aka .NET Native for Universal Windows Apps), this applies to a few performance critical helpers in the class library that are known to never call through a fat function pointer. We can just file an issue to track it.